miniapp/miniapp_triangular_multiplication.cpp

```diff
@@ -0,0 +1,286 @@
+            comm_grid, side, uplo, op, diag, alpha, a.get(), b.get());
+
+      sync_barrier();
+
+      // benchmark results
+      if (0 == world.rank() && run_index >= 0) {
+        auto elapsed_time = timeit.elapsed();
+        double gigaflops = total_ops / elapsed_time / 1e9;
+
+        std::cout << "[" << run_index << "]"
+                  << " " << elapsed_time << "s"
+                  << " " << gigaflops << "GFlop/s"
+                  << " " << dlaf::internal::FormatShort{opts.type}
+                  << dlaf::internal::FormatShort{opts.side} << dlaf::internal::FormatShort{opts.uplo}
+                  << dlaf::internal::FormatShort{opts.op} << dlaf::internal::FormatShort{opts.diag}
+                  << " " << bh.size() << " " << bh.blockSize() << " " << comm_grid.size() << " "
+                  << pika::get_os_thread_count() << " " << backend << std::endl;
+      }
+
+      b.copyTargetToSource();
+
+      // (optional) run test
+      if ((opts.do_check == dlaf::miniapp::CheckIterFreq::Last && run_index == (opts.nruns - 1)) ||
+          opts.do_check == dlaf::miniapp::CheckIterFreq::All) {
+        DLAF_UNIMPLEMENTED("Check");
+      }
+    }
+  }
+};
+
+int pika_main(pika::program_options::variables_map& vm) {
+  pika::scoped_finalize pika_finalizer;
+  dlaf::ScopedInitializer init(vm);
+
+  const Options opts(vm);
+  dlaf::miniapp::dispatchMiniapp<triangularMultiplicationMiniapp>(opts);
+
+  return EXIT_SUCCESS;
+}
+
+int main(int argc, char** argv) {
+  dlaf::comm::mpi_init mpi_initter(argc, argv);
+
+  // options
+  using namespace pika::program_options;
+  options_description desc_commandline(
+      "Benchmark computation of A . B, "
+      "where A is a non-unit lower triangular matrix, and B is an m by n matrix\n\n"
+      "options\n"
+      "Usage: miniapp_triangular_multiplication [options]");
+  desc_commandline.add(dlaf::miniapp::getMiniappOptionsDescription());
+  desc_commandline.add(dlaf::getOptionsDescription());
+
+  // clang-format off
+  desc_commandline.add_options()
+    ("m",             value<SizeType>()     ->default_value(4096),       "Matrix b rows")
+    ("n",             value<SizeType>()     ->default_value(512),        "Matrix b columns")
+    ("mb",            value<SizeType>()     ->default_value(256),        "Matrix b block rows")
+    ("nb",            value<SizeType>()     ->default_value(512),        "Matrix b block columns")
+  ;
+  // clang-format on
+  dlaf::miniapp::addSideOption(desc_commandline);
+  dlaf::miniapp::addUploOption(desc_commandline);
+  dlaf::miniapp::addOpOption(desc_commandline);
+  dlaf::miniapp::addDiagOption(desc_commandline);
+
+  pika::init_params p;
+  p.desc_cmdline = desc_commandline;
+  p.rp_callback = dlaf::initResourcePartitionerHandler;
+  return pika::init(pika_main, argc, argv, p);
+}
+
+namespace {
+/// Returns a tuple of element generators of three matrices A(m x m), B (m x n), X (m x n), for which it
+/// holds op(A) X = alpha B (alpha can be any value).
```

Comment:
```suggestion
/// holds alpha op(A) B = X (alpha can be any value).
```
I know this will disappear, but maybe it's worth to change the naming convention to be clearer/more consistent with ScaLAPACK `trmm`? (See potential confusion between `ref_b` and `ref_x`.)

Reply:
I don't find it confusing. It is the same notation we use in tests.
https://github.com/eth-cscs/DLA-Future/blob/master/test/include/dlaf_test/matrix/util_generic_blas.h#L236

Reply:
We have the same mismatch here: https://github.com/eth-cscs/DLA-Future/blob/master/test/unit/test_blas_tile/test_trmm.h#L47-L48
The only difference it that we call the variables differently.

Would it be ok with you to just change https://github.com/eth-cscs/DLA-Future/pull/1029/files#diff-299ea192c66316a4824dc3429626878865488909cfb9639128d5e032d90158d3R128 to
`auto [in_op_a, out_b, in_b] = ...`
and update the sets?

Reply:
Yes, this sounds reasonable and clearer.